lib/spack/spack/test/config_requirements.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+import pytest
+
+import spack.config
+import spack.repo
+import spack.util.spack_yaml as syaml
+from spack.solver.asp import UnsatisfiableSpecError
+from spack.spec import Spec
+
+pytestmark = pytest.mark.skipif(sys.platform == "win32", reason="Windows uses old concretizer")
+
+
+def update_packages_config(conf_str):
+    conf = syaml.load_config(conf_str)
+    spack.config.set("packages", conf["packages"], scope="concretize")
+
+
+_pkgx = (
+    "x",
+    """\
+class X(Package):
+    version('1.1')
+    version('1.0')
+    version('0.9')
+
+    variant('shared', default=True,
+            description='Build shared libraries')
+
+    depends_on('y')
+""",
+)
```

Comment:
(comment) I wonder if we can generalize this, and use Jinja2 and templates. Anyhow, this is for sure something for after this PR.